src/main/java/org/demo/service/cxbox/inner/DictionaryService.java

```diff
@@ -0,0 +1,142 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.demo.service.cxbox.inner;
+
+import jakarta.persistence.EntityManager;
+import java.util.stream.Collectors;
+import org.cxbox.api.data.dictionary.DictionaryCache;
+import org.cxbox.api.data.dto.DataResponseDTO;
+import org.cxbox.api.service.LocaleService;
+import org.cxbox.core.crudma.bc.BusinessComponent;
+import org.cxbox.core.crudma.impl.VersionAwareResponseService;
+import org.cxbox.core.dto.BusinessError.Entity;
+import org.cxbox.core.dto.rowmeta.ActionResultDTO;
+import org.cxbox.core.dto.rowmeta.CreateResult;
+import org.cxbox.core.exception.BusinessException;
+import org.cxbox.core.service.action.ActionScope;
+import org.cxbox.core.service.action.Actions;
+import org.cxbox.model.core.api.TranslationId;
+import org.cxbox.model.dictionary.entity.DictionaryItem;
+import org.cxbox.model.dictionary.entity.DictionaryItemTranslation;
+import org.cxbox.model.dictionary.entity.DictionaryItem_;
+import org.cxbox.model.dictionary.entity.DictionaryTypeDesc;
+import org.demo.dto.cxbox.inner.DictionaryItemDTO;
+import org.demo.dto.cxbox.inner.DictionaryItemDTO_;
+import org.demo.repository.DictionaryRepository;
+import org.hibernate.exception.ConstraintViolationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+@SuppressWarnings({"java:S6813"})
+@Service
+public class DictionaryService extends VersionAwareResponseService<DictionaryItemDTO, DictionaryItem> {
+
+	@Autowired
+	private DictionaryCache dictionaryCache;
+
+	@Autowired
+	private DictionaryRepository dictionaryRepository;
+
+	@Autowired
+	private EntityManager entityManager;
+
+	@Autowired
+	private LocaleService localeService;
+
+	public DictionaryService() {
+		super(DictionaryItemDTO.class, DictionaryItem.class, null, DictionaryMeta.class);
+	}
+
+	@Override
+	protected ActionResultDTO<DictionaryItemDTO> doUpdateEntity(DictionaryItem entity, DictionaryItemDTO data,
+			BusinessComponent bc) {
+		setIfChanged(data, DictionaryItemDTO_.type, entity::setType);
+		if (data.isFieldChanged(DictionaryItemDTO_.dictionaryTypeId)) {
+			entity.setDictionaryTypeId(data.getDictionaryTypeId() != null
+					? entityManager.getReference(DictionaryTypeDesc.class, data.getDictionaryTypeId())
+					: null);
+		}
+		setIfChanged(data, DictionaryItemDTO_.key, entity::setKey);
+		setIfChanged(data, DictionaryItemDTO_.value, entity::setValue);
+		if (data.isFieldChanged(DictionaryItemDTO_.value)) {
```

Comment:
twice for field DictionaryItemDTO_.value

Reply:
1) dictionary_item
2) dictionary_item_tr (translations)